packages/core/src/internal/verifyTransaction/rebuildTransactionPostData.ts

```diff
@@ -0,0 +1,327 @@
+import BigNumber from 'bignumber.js';
+
+interface BaseTransaction {
+    type?: number;
+    version?: number;
+    subtype?: number;
+    timestamp?: number;
+    deadline?: number;
+    senderPublicKey?: string;
+    sender?: string;
+    recipient?: string;
+    amountNQT?: string;
+    feeNQT?: string;
+    referencedTransactionFullHash?: string;
+    signature?: string;
+    flags?: number;
+    ecBlockHeight?: number;
+    ecBlockId?: string;
+    cashBackId?: string;
+}
+
+interface AttachmentFields {
+    type: string;
+    parameterName?: string;
+}
+
+interface AttachmentSpec {
+    request: string;
+    fields: AttachmentFields[];
+}
+
+/**
+ * Mini class to easily implement reading data sequentially.
+ */
+class ByteBuffer {
```

Comment:
Better move this into a separate file and do unit tests for each function...

Reply:
Done.